app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java

```diff
@@ -294,117 +285,114 @@ public static boolean almostEquals( TransmitterRawData e1, TransmitterRawData e2
                 TransmitterRawData trd = gson.fromJson(data, TransmitterRawData.class);
                 trd.CaptureDateTime = System.currentTimeMillis() - trd.RelativeTime;
 
-                trd_list.add(0,trd);
+                trd_list.add(0, trd);
                 //  System.out.println( trd.toTableString());
-                if(trd_list.size() == numberOfRecords) {
-                	// We have the data we want, let's get out
-                	break;
+                if (trd_list.size() == numberOfRecords) {
+                    // We have the data we want, let's get out
+                    break;
                 }
             }
 
 
-            MySocket.close();
+            mySocket.close();
             return trd_list;
-        }catch(SocketTimeoutException s) {
+        } catch (SocketTimeoutException s) {
             Log.e(TAG, "Socket timed out! ", s);
-        }
-        catch(IOException e) {
+        } catch (IOException e) {
             Log.e(TAG, "cought IOException! ", e);
         }
         return trd_list;
     }
 
 
-    public void run()
-    {
-    	Long LastReportedTime = new Date().getTime();
-    	TransmitterRawData LastReportedReading = null;
-    	Log.e(TAG, "Starting... LastReportedReading " + LastReportedReading);
-    	try {
-	        while (!mStop && !interrupted()) {
-	        	// try to read one object...
-                TransmitterRawData[] LastReadingArr = null;
-                if(WixelReader.IsConfigured(mContext)) {
+    public void run() {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method run() has an NPath complexity of 15](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l308)